app/ide-desktop/lib/dashboard/src/hooks/useOnScroll.ts

```diff
@@ -0,0 +1,43 @@
+/** @file Execute a function on scroll. */
+import * as React from 'react'
+
+// ===================
+// === useOnScroll ===
+// ===================
+
+/** Execute a function on scroll. */
+export default function useOnScroll(callback: () => void, dependencies: React.DependencyList) {
+  const callbackRef = React.useRef(callback)
```

Comment:
btw, we have `useEventCallback` app/ide-desktop/lib/dashboard/src/hooks/eventCallbackHooks.ts that does exactly this.